automator/src/com/testsigma/automator/actions/mobile/android/generic/ChangeScreenOrientationAction.java

```diff
@@ -1,23 +1,26 @@
 package com.testsigma.automator.actions.mobile.android.generic;
 
+import com.google.common.collect.ImmutableMap;
+import com.testsigma.automator.actions.mobile.generic.GetScreenOrientationAction;
 import org.openqa.selenium.ScreenOrientation;
+import org.openqa.selenium.remote.DriverCommand;
 
 public class ChangeScreenOrientationAction extends com.testsigma.automator.actions.mobile.generic.ChangeScreenOrientationAction {
 
   public void changeOrientation() {
-    if (getDriver().getOrientation().equals(ScreenOrientation.LANDSCAPE)) {
-      changeToPortrait();
-    } else {
-      changeToLandscape();
-    }
+    if (getDriver().execute(DriverCommand.GET_SCREEN_ORIENTATION).getValue().toString().equalsIgnoreCase(ScreenOrientation.LANDSCAPE.value())) {
```

Comment:
getting GET_SCREEN_ORIENTATION value if it returns null we might have issue. I remember we had similar issues in cloud as well initially. Can you please reverse the comparision and also have some default value in case we are not getting any.

Reply:
Addressed